### DISCUSSION TITLE
Random Error 500: Internal Server Error

### DISCUSSION THREAD

**Discussion by DanCarl857** [vercel/community#6535] [Help]
### Summary

We have a deployed nextjs (v13) application on vercel. Our issue is we are consistently seeing 500 errors from various pages and from varying actions being carried out in the application. These errors seem to self-correct after some time and are not being caught by our error handlers in the application which leads us to believe the app is failing before it is even rendered. Now a ton of these errors are not reproducible because as I mentioned they seem to self-correct almost immediately but it's becoming a bit too frequent for our users. The logs don't give us much as all we get is `Internal Server error`, any ideas as to what could be the cause of this and possible fixes? Also any ideas as to how we can get more definitive errors?

### Example
<img width="534" alt="Screenshot 2024-04-17 at 06 46 13" src="https://github.com/vercel/community/assets/9010142/0d2682bd-f4d3-46ad-a5fa-78e2a2c9fe5b">
 
 This is an example of one such errors. Here I am trying to go to the inbox page. No failing API calls, nothing in the console, just a 500 screen

### Steps to Reproduce

Difficult to reproduce at the moment seeing as these errors are random

**Comment by nabiedma** [comment]
Hey there! I am having the same issue. When deploying the app and testing it (in a dev branch for example) at the start it fails in one of our endpoints with this 500 Internal Server Error. If I reload the page, it start working ok with no errors. But later, with no reason, other endpoints send the same response, a 500 Internal Server Error and the app crases. It's really annoying and the logs are not helpful.

**Reply by Reniactor** [reply]
Having the exact same issue here

**Reply by nabiedma** [reply]
Hey @Reniactor! Yesterday I created a new discussion. I have detailed a little bit more about the problem and issues I am having. If you could take a look or maybe leave a reply there with your problem, that would be great! Thanks and I hope we can solve this 😞 
https://github.com/orgs/vercel/discussions/6828

**Reply by Reniactor** [reply]
Hey man, sorry i didn't reply. I investigated and, in my case, i just had to add this line: ?pgbouncer=true at the end of my database url in vercel. I hope this helps you out too
